contract/models/contract_line.py

```diff
@@ -1164,17 +1164,33 @@ def _get_renewal_dates(self):
         )
         return date_start, date_end
 
+    @api.multi
+    def _renew_create_line(self, date_start, date_end):
+        self.ensure_one()
+        is_auto_renew = self.is_auto_renew
+        self.stop(self.date_end, post_message=False)
+        new_line = self.plan_successor(
+            date_start, date_end, is_auto_renew, post_message=False
+        )
+        new_line._onchange_date_start()
+        return new_line
+
+    @api.multi
+    def _renew_extend_line(self, date_end):
+        self.ensure_one()
+        self.date_end = date_end
+        return self
+
     @api.multi
     def renew(self):
         res = self.env['contract.line']
         for rec in self:
-            is_auto_renew = rec.is_auto_renew
-            rec.stop(rec.date_end, post_message=False)
+            company = rec.contract_id.company_id
             date_start, date_end = rec._get_renewal_dates()
-            new_line = rec.plan_successor(
-                date_start, date_end, is_auto_renew, post_message=False
-            )
-            new_line._onchange_date_start()
+            if company.create_new_line_at_contract_line_renew:
+                new_line = rec._renew_create_line(date_start, date_end)
+            else:
+                new_line = rec._renew_extend_line(date_end)
```

Comment:
It bothers me that one method takes `date_start` and the other not. It should be symmetric or bugs will creep in later. Since it does not make sense for `_renew_extend_line` to take `date_start` then it must be removed from `_renew_create_line` (in which case it is always date_end + 1 day), and `_get_renewal_dates` must be changed to `_get_renewal_new_date_end`.

Reply:
make sense